joomla/joomla-cms
# Prevent create categories in first level if user is not allowed

This PR will fix a part of the issues described in #8069 
## The issue

A user with ACL right to create, delete, edit, edit State, edit own in a specified category can also create new categories on the first level. (no parent)
## How to test
1. Create a usergroup with registered as parent
2. Select that group in the "Special" view/access level
3. In Global configuration, allow Administrator Login for this group
4. In the Global options of Articles, Categories and Featured Articles set "Access Administrator interface" to <i>Allowd</i> for that group
5. Create a new category and set the permissions of hat categories for the new user group for the following item on <i>allowed</i>:
   - Create
   - Delete
   - Edit
   - Edit state
   - Edit Own
6. Create a new user for the created user group
7. Login with the user on the back-end
8. Click on the create button and notice you can create a category for "no parent"
9. Apply patch, and notice you can't create categories for "no parent" any more
## Note

Can someone please review if this is a good solution? I know it solves the issue, but I don't know exactly why that piece of code was placed
